mesh/mesh.cpp

```diff
@@ -7913,6 +7913,9 @@ void Mesh::SetNodes(const Vector &node_coord)
    {
       SetVertices(node_coord);
    }
+
+   // Invalidate the old geometric factors
+   DeleteGeometricFactors();
```

Comment:
Couldn't we call `Update` here and for the other methods updating the nodes?

Reply:
I was debating on this, but you are probably right. I can switch them to `Update`

Reply:
Done in https://github.com/mfem/mfem/pull/3235/commits/9f3974d7c3d04306a679ced981f27ebd8cc30c9d